File: cypress-benchmark/cypress/support/do-view-check-wait/check.ts
/**
 * Check: encapsulate all generic web-page assert methods
*/
export class Check {
    /**
     * Assert that a certain number of elements are present on the page.
     * @param locator The locator of the elements to query.
     * @param expectedLength The expected length of elements.
     * @param timeout The timeout to get the elements, default to 30s.
    */
    public haveLength(locator: string, expectedLength: number, timeout = 30): void {
        cy.get(locator, {timeout: timeout * 1000}).should('have.length', expectedLength);
    }

    /**
     * Assert that a certain number of elements are present on the page.
     * @param locator The locator of the elements to query.
     * @param expectedText The expected text of the given element.
     * @param index (Optional) The index of the element, default 0
     * @param timeout (Optional) The timeout to get the elements, default to 30s.
    */
    public haveText(locator: string, expectedText: string, index: number = 0, timeout = 30): void {
        cy.get(locator, {timeout: timeout * 1000}).eq(index).should('have.text', expectedText);
    }

    /**
     * Assert that a certain element is visible.
     * @param locator The locator of the elements to query.
     * @param timeout The timeout to get the elements, default to 30s
    */
    public visible(locator: string, timeout = 30) {
        cy.get(locator, { timeout: timeout * 1000 }).should('be.visible');
    }
}

export default new Check();

File: playwright_py_benchmark/layer0_automation_tool/do_view_check/playwright/check.py
from playwright.sync_api import expect
from .dvc import DoViewCheck

class Check(DoViewCheck):
    
    def visibility(locator: str, timeout = 60) -> None:
        """_summary_
        Assert that the element is visible within specified timeout.
        Args:
            locator (str): _description_
            timeout (int, optional): _description_. Defaults to 60.
        """
        expect(Check.page.locator(locator)).to_be_visible(timeout=timeout * 1000)
    
    def text(locator: str, expectedText: str, timeout = 60) -> None:
        """_summary_

        Args:
            locator (str): _description_
            expectedText (str): _description_
            timeout (int, optional): _description_. Defaults to 60.
        """
        expect(Check.page.locator(locator)).to_contain_text(expected=expectedText, timeout=timeout * 1000)